conan/tools/sbom/cyclonedx.py

```diff
@@ -211,10 +215,15 @@ def _calculate_licenses(component):
     ]
 
 
-def _calculate_bomref(component):
+def _calculate_bomref(component, qualifiers):
     user = f"&user={component.ref.user}" if component.ref.user else ""
     channel = f"&channel={component.ref.channel}" if component.ref.channel else ""
-    return f"pkg:conan/{component.name}@{component.ref.version}?rref={component.ref.revision}{user}{channel}"
+    purl_qualifier = ""
+    if hasattr(component, "conanfile"):
+        purl_qualifier = "".join(f"&{qualifier}={component.conanfile.settings.get_safe(qualifier)}" for qualifier in qualifiers)
+    return (f"pkg:conan/{component.name}@{component.ref.version}"
+            f"?rref={component.ref.revision}&pref={component.pref.package_id}"
```

Comment:
We have discussed that this should be rrev, but we have decided to do that in a different PR